tests/test_tokens.py

```diff
@@ -60,3 +80,87 @@ def test_token_process_with_valid_action(self, mock_handler):
                 self.encoded_token
             )
         )
+
+
+SANCTION_CLASS_MAP = {
+    'embargo': Embargo,
+    'registration_approval': RegistrationApproval,
+    'retraction': Retraction
+}
+SANCTION_FACTORY_MAP = {
+    'embargo': factories.EmbargoFactory,
+    'registration_approval': factories.RegistrationApprovalFactory,
+    'retraction': factories.RetractionFactory
+}
+
+def make_sanction_token_handler_case(kind):
```

Comment:
Discussed over Flowdock:

This makes test selection, readability, and debugging difficult. I think using a base test class with helper methods would be an improvement.